pycbc/events/single.py

```diff
@@ -1,61 +1,152 @@
+                                 'Default conservative. Can be given as '
+                                 'a single value or as detector-value pairs, '
+                                 'e.g. H1:mean L1:mean V1:conservative')
 
     @classmethod
     def from_cli(cls, args, ifo):
         return cls(
            ifo, newsnr_threshold=args.single_newsnr_threshold[ifo],
            reduced_chisq_threshold=args.single_reduced_chisq_threshold[ifo],
-           fixed_ifar=args.single_fixed_ifar[ifo],
            duration_threshold=args.single_duration_threshold[ifo],
+           fixed_ifar=args.single_fixed_ifar,
+           fit_file=args.single_fit_file,
+           sngl_ifar_est_dist=args.sngl_ifar_est_dist[ifo]
            )
 
-    def check(self, triggers, data_reader):
+    def check(self, trigs, data_reader):
         """ Look for a single detector trigger that passes the thresholds in
         the current data.
         """
-        if len(triggers['snr']) == 0:
+
+        if len(trigs['snr']) == 0:
+            return None
+
+        # Apply cuts to trigs before clustering
+        # Cut on snr so that triggers which could not reach newsnr
+        # threshold do not have newsnr calculated
+        valid_idx = (trigs['template_duration'] >
+                     self.thresholds['duration']) & \
+                    (trigs['chisq'] <
+                     self.thresholds['reduced_chisq']) & \
+                    (trigs['snr'] >
+                     self.thresholds['newsnr'])
+        if not np.any(valid_idx):
+            return None
+
+        cutdurchi_trigs = {k: trigs[k][valid_idx] for k in trigs}
+        nsnr_all = ranking.newsnr(cutdurchi_trigs['snr'],
+                                  cutdurchi_trigs['chisq'])
+        nsnr_idx = nsnr_all > self.thresholds['newsnr']
+        if not np.any(nsnr_idx):
             return None
 
-        i = triggers['snr'].argmax()
+        cutall_trigs = {k: cutdurchi_trigs[k][nsnr_idx]
+                        for k in trigs}
+
+        # 'cluster' by taking the maximal newsnr value over the trigger set
+        i = nsnr_all[nsnr_idx].argmax()
+
         # This uses the pycbc live convention of chisq always meaning the
         # reduced chisq.
-        rchisq = triggers['chisq'][i]
-        nsnr = ranking.newsnr(triggers['snr'][i], rchisq)
-        dur = triggers['template_duration'][i]
-
-        if nsnr > self.newsnr_threshold and \
-                rchisq < self.reduced_chisq_threshold and \
-                dur > self.duration_threshold:
-            fake_coinc = {'foreground/%s/%s' % (self.ifo, k): triggers[k][i]
-                          for k in triggers}
-            fake_coinc['foreground/stat'] = nsnr
-            fake_coinc['foreground/ifar'] = self.fixed_ifar
-            fake_coinc['HWINJ'] = data_reader.near_hwinj()
-            return fake_coinc
-        return None
+        rchisq = cutall_trigs['chisq'][i]
+        nsnr = ranking.newsnr(cutall_trigs['snr'][i], rchisq)
```

Comment:
Looks like you could recycle the previous NewSNR calculation, and replace the above two lines with
```python
nsnr = nsnr_all[nsnr_idx][i]
```
Really just aesthetics though, up to you.